### DISCUSSION TITLE
[Esame di prova 30 dicembre 2023] Esercizio 3

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/algebra#388] [Esami - Viaggi/Piazza] [answered]
![immagine](https://github.com/sapienzastudentsnetwork/algebra2324/assets/107944476/b4953ab6-b6b7-4735-8f21-cbd6e97f7fba)

**Comment by IonutCicio** [comment]
## (1)

### $f$ è iniettiva $\implies H \cap K = \\{ 1_G \\}$

Supponiamo per assurdo che $H \cap K \neq \\{ 1_G \\}$ vuol dire che esiste $g \in G$ tale che $g \in H \land g \in K$ quindi sono definiti $f((g, 1_G)) = g \cdot 1_G = g = 1_G \cdot g = f((1_G, g))$ che è una contraddizione all'ipotesi che $f$ è iniettiva

> [!NOTE]
Passare al complemento per questa implicazione potrebbe essere più comodo

### $H \cap K \neq \\{ 1_G \\} \implies f$ non è iniettiva

Infatti esiste $g \in G$ tale che $g \in H \land g \in K$ quindi sono definiti $f((g, 1_G)) = g \cdot 1_G = g = 1_G \cdot g = f((1_G, g))$ per cui $f$ non è iniettiva

###  $H \cap K = \\{ 1_G \\} \implies f$ è iniettiva

Supponiamo per assurdo che $f$ non sia iniettiva, $f((h, k)) = hk = h'k' = f((h', k'))$ ma $hk = h'k' \implies h'k'(hk)^{-1} = 1_G \implies h'k'k^{-1}h^{-1} = 1_G \implies k' = h \land h' = k \implies h' \in H \land h' \in K \land k' \in H \land k' in K$ che è contro l'ipotesi che $H \cap K = \\{ 1_G \\}$

## (2)

### $\forall h \in H, \forall k \in K \quad h \cdot k = k \cdot h \implies f$ è un omomorfismo

Si ha $f((h, k) \cdot (h', k')) = f((hh', kk')) = hh'k k' \text{ (per ipotesi)} = hkh'k'  = f((h, k)) \cdot f((h', k'))$

### $f$ è un omomorfismo $\implies \forall h \in H, \forall k \in K \quad hk = kh$

Supponimo per assurdo che $f$ è un omomorfismo e $\exists h' \in H, k \in K : h'k \neq kh'$, per ipotesi abbiamo

- $f((h, k) \cdot (h', k')) = f((hh', kk')) = hh'kk'$
- $f((h, k)) \cdot f((h', k')) = hkh'k'$

Ma $hh'kk' \neq hkh'k'$ che è una contraddizione rispetto all'ipotesi che $f$ sia un omomorfismo

**Reply by cornflexxx** [reply]
Nel punto (1) quando parti da ($H\cap K =\\{1_G\\}\Rightarrow \text{f è iniettiva}$) potevi anche vedere che:
Supponi per assurdo che $f$ non è iniettiva, quindi $(h,k)\neq(h',k')$ , $f(h,k)=f(h,'k')\Rightarrow hk=h'k'$, ma allora $h'k'(hk)^{-1}=1_G\Longleftrightarrow h'k'k^{-1}h^{-1}=1_G\Longleftrightarrow h=h' \wedge k=k'$, quindi $(h,k)=(h',k')$ in contrasto con la nostra ipotesi , quindi $f$ è iniettiva.

**Reply by IonutCicio** [reply]
Intendi che bastava usare direttamente $\iff$ senza dimostrare entrambi i versi?

**Reply by cornflexxx** [reply]
Non l’avevo pensata in questi termini io dicevo solo che $(hk)^{-1}=k^{-1}h^{-1}$ e non $h^{-1}k^{-1}$ se il gruppo non è commutativo .

**Reply by IonutCicio** [reply]
Vero, non avevo notato che avevi invertito $h$ e $k$. Questo perché $hk = h'k' \implies hkk^{-1} = h'k'k^{-1} \implies h = h'k'k^{-1} \implies hh^{-1} = h'k'k^{-1}h^{-1} \implies 1_G = h'k'k^{-1}h^{-1}$. _(correggo anche nella soluzione)_

**Comment by CiottoloMaggico** [comment]
![image](https://github.com/sapienzastudentsnetwork/algebra2324/assets/79638284/c67959b7-5ab6-4377-8a8f-f629a9e61fca)

**Comment by Elia-Belli** [comment, marked as answer]
## Soluzione Ufficiale
![immagine](https://github.com/sapienzastudentsnetwork/algebra2324/assets/107944476/56316b21-7d1f-4bf3-a9fa-6496e8d0511b)

**Reply by Elia-Belli** [reply]
**(a)** Potrebbe non essere evidente e quindi preciso, riguardo alla dimostrazione $f$ non-iniettiva $\Rightarrow H\cap K\neq \\{1_G\\}$:

Dopo aver assunto $x\neq h$ moltiplichiamo ambo i lati dell'uguaglianza, a sinistra per $h^{-1}$ e a destra per $y^{-1}$:

$$xy=hk\to h^{-1}xyy^{-1}=h^{-1}hky^{-1}\to h^{-1}x=ky^{-1}$$

Da qui abbiamo che $(h^{-1}x)\in H$ e $(ky^{-1})\in K$ perchè il prodotto di due elementi di un gruppo è ancora nel gruppo.
Infine essendo i prodotti uguali sappiamo che $h^{-1}x=ky^{-1}\in H\cap K$

> L'ipotesi di $x\neq h$ serve affinchè $h^{-1}x\neq 1_G$, si ottiene lo stesso risultato ponendo $y\neq k$ oppure $y\neq k \ \land\ x\neq h$
